File: AndroidApplication/app/src/main/java/com/crossp/client/Ad.java
package com.crossp.client;

public class Ad {
    private String externalId;
    public String getExternalId() { return externalId; }

    private String name;
    public String getName() { return name; }

    private String url;
    public String getUrl() { return url; }

    private String projectExternalId;
    public String getProjectExternalId() { return projectExternalId; }

    private String fileExternalId;
    public String getFileExternalId() { return fileExternalId; }

    {
        externalId = null;
        name = null;
        url = null;
        projectExternalId = null;
        fileExternalId = null;
    }

    public Ad(String externalId, String name, String url, String projectExternalId, String fileExternalId) {
        this.externalId = externalId;
        this.name = name;
        this.url = url;
        this.projectExternalId = projectExternalId;
        this.fileExternalId = fileExternalId;
    }
}

File: WebApplication/Ads/Ads/Models/Ad.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace Ads.Models
{
    public class Ad
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; } = default;
        public string ExternalId { get; set; }

        public string Name { get; set; }

        public string Url { get; set; }

        public string ProjectExternalId { get; set; } = default;
        
        public string FileExternalId { get; set; } = default;

        public Ad() { }
        public Ad(string projectExternalId, string fileExternalId)
        {
            if (string.IsNullOrWhiteSpace(projectExternalId))
                throw new NullReferenceException($"Parameter {nameof(projectExternalId)} passed into {nameof(Ad)} constructor is null or empty.");

            if (string.IsNullOrWhiteSpace(fileExternalId))
                throw new NullReferenceException($"Parameter {nameof(fileExternalId)} passed into {nameof(Ad)} constructor is null or empty.");

            ProjectExternalId = projectExternalId;
            FileExternalId = fileExternalId;
            ExternalId = Guid.NewGuid().ToString();
            Name = "New ad";
            Url = "Empty";
        }
    }
}
